firestore/integration_test.go

```diff
@@ -681,7 +681,91 @@ func TestIntegration_Query(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	seen := map[int64]bool{} // "q" values we see
+	seen := map[int64]bool{} // "q" values we see.
+	for _, d := range allDocs {
+		data := d.Data()
+		q, ok := data["q"]
+		if !ok {
+			// A document from another test.
+			continue
+		}
+		if seen[q.(int64)] {
+			t.Errorf("%v: duplicate doc", data)
+		}
+		seen[q.(int64)] = true
+		if data["x"] != int64(1) {
+			t.Errorf("%v: wrong or missing 'x'", data)
+		}
+		if len(data) != 2 {
+			t.Errorf("%v: want two keys", data)
+		}
+	}
+	if got, want := len(seen), len(wants); got != want {
+		t.Errorf("got %d docs with 'q', want %d", len(seen), len(wants))
+	}
+}
+
+func TestIntegration_QueryDocuments_LimitToLast_Fail(t *testing.T) {
+	ctx := context.Background()
+	coll := integrationColl(t)
+	q := coll.Select("q").OrderBy("q", Asc).LimitToLast(1)
+	gotDocs, err := q.Documents(ctx).GetAll()
+	if err == nil {
+		t.Errorf("got %v docs, want error", len(gotDocs))
+	}
+}
+
+func TestIntegration_QueryGet(t *testing.T) {
```

Comment:
This test is a complete analogue of `TestIntegration_QueryDocuments`, but with `Get()` method and `LimitToLast()` test case.